### DISCUSSION TITLE
How to use the d-sharp/e-flat of the Norrfjärden organ?

### DISCUSSION THREAD

**Discussion by kerkovits** [GrandOrgue/grandorgue#1379] [Help] [answered]
@larspalo The Norrfjärden organ has separate keys for d# and eb. I have a regular keyboard attached to the full compass manuals. How can I tell GO that I want to use d# instead of the default eb (of course, in all octaves)? (i.e. I could switch between e minor and b flat major). The method does not necessarily have to be manageable by MIDI events, I am fine if I could set it by mouse, as one rarely needs d# and eb in the same piece.

**Comment by larspalo** [comment]
@kerkovits I admit that I never thought about that! Currently the d# tones are implemented exclusively on their own manuals which otherwise only contain DUMMY pipes. I even admit that I didn't bother to tune the d# notes as I thought they were there more for the authentic look than for actual usage...

Ideally, a solution should still visibly play the other keys and only differentiate between if the d# or eb is played.

I'll have to test a few ideas, but feel free to test it yourself and if you find a solution that works please report back! If it's something that requires changing the odf, I can eventually release a new version where I also have tuned the d# pipes properly too...

**Reply by kerkovits** [reply]
Thanks, @larspalo. I just played a piece in d major, and everything sounded perfect (much better than in an equal temperament), until a VI#6 (b major/d#) chord. The e flat instead of the d sharp in the bass was horrific. :smile: My first idea was to change the ODF, reference all pipes of the normal rank in the subsemitones rank (except the d#) and use an 8 and an UO coupler. However, I could not even use the UO coupler, as all pipes sounded regardless of the UO activated.

**Comment by kerkovits** [comment]
@larspalo At last, I had a little time to dig into this problem. The proposed patch:

- Replaces DUMMY pipes with REF pipes, so all ranks are complete. If both the normal and the subsemitone stops are engaged, the pipe sounds only once (except the D#/Eb).
- Removes the unnecessary subsemitones couplers (resolved by the REF pipes).
- Adds new couplers, so that the I/P, II/P and III/II are effective between subsemitones manuals.
- Adds a new switch Eb->D# to the stops panel. If engaged, it disengages all normal stops, and couples the full compass manual to the subsemitones manual.

What it does:

- Using the original manuals, everything is unchanged, but all keys of the subsemitones manuals will sound.
- The full compass manuals get correctly mapped to the main panel. Normally, an Eb sound is heard. If one activates Eb->D#, the subseminones D# is pressed on the main panel (and it is heard instead of D#).

Problems:

- The Sperrventil is ineffective on the subsemitones manuals (this bug has already existed before).
- Minor visual glitch: With Eb->D# activated, the Eb key of the main panel seems to be pressed (but does not sound, so not a signficant problem).
- Connecting a manual to the main panel manuals and activating Eb->D# at the same time will result in no sound output. (However, the main panel manuals do not need the Eb->D# switch at all, as D# can be accessed from a separate manual.
- The Eb->D# switch should be somehow accessible on the panel of full compass manuals, too.

Nevertheless, it is a proof concept. I can play on it and the D# pipes now sounds perfect in E minor.

(for some reason, GitHub did not allowed to use the *.patch extension)
[norrfjarden.txt](https://github.com/GrandOrgue/grandorgue/files/11183227/norrfjarden.txt)

**Reply by larspalo** [reply]
@kerkovits I'm thinking that something like this might work. 
[BothDisAndEb.zip](https://github.com/GrandOrgue/grandorgue/files/11184367/BothDisAndEb.zip)

Uppermost manual is the Subsemitone manual with REFs like you though.
Middle manual is the full compass manual
Lower manual is the original manual

The trick coupler(s) is for transferring the key presses on the subsemitone manual to the original manual except for MIDI keys 051, 063 and 075. This trick coupler (actually a switch that activates four couplers) would be constantly active. Also only the three D# keys of the subsemitone manual would actually be visible.

This could work so that when the user connects to the original keyboard the Eb would play. When the user connects to the D# it would actually be to the Subsemitone manual instead. Switching between using Eb and D# would not be so comfortable perhaps, but it would look perfect on the gui panel.

**Reply by larspalo** [reply]
@kerkovits This version is almost perfect in my opinion.
[UseBothDisAndEb2.zip](https://github.com/GrandOrgue/grandorgue/files/11191281/UseBothDisAndEb2.zip)

Unzip and drop into the sample set folder of Norrfjärden.

There are a few options for the user:

- One can connect to  original manual (uses short octave and Eb).
- One can connect to subsemitone manual (uses full range 036-084 and D#)
- One can connect to the modern keyboard full compass on its own panel and switch between using Eb or D#. (If D# is chosen the range will currently be limited to 084, but that could easily be changed)

The correct notes on the main panel keyboards will play in any case (if they exist there).

If the subsemitone manual would be extended in range and another D# (087) would be provided (needs to be made as it's not existing in the original instrument) one could play the whole modern range and freely switch between Eb and D# on the fly.

**Reply by kerkovits** [reply]
@larspalo Sorry that I had time only now to test the solution. It works as expected. Good work, thank you!

**Reply by larspalo** [reply]
@kerkovits So, you think that I should implement it like this for then next update? I also have received a suggestion to have more plain labels for the modern keyboard console so that it should be easier to read. I also have fixed a few samples that had incorrect pitch info embedded.

**Reply by kerkovits** [reply]
Yes, I do think this.

I personally prefer these authentic scripts for stop labels. I would not recommend to change them significantly. Maybe some contrast optimization would help in increasing readability while retaining the look&feel.

**Reply by larspalo** [reply]
Great, I'll do that eb/d# modification then for the whole sample set!

My intention was to keep the original main panel as is (at least until next time I can visit Norrfjärden and make better images of the labels than the ones I already have with all the shadows...), but possibly extract one label without any hand written text on and instead write out the stop names in plain text from GO on them for the Modern full compass panel only, so that it would be easier to read. This comes from a suggestion Mark sent to me in an e-mail.

Maybe I should add additional panels for the left jamb and right jamb too?

**Reply by kerkovits** [reply]
It should be great.

The exact layout of panels is completely up to your discretion. I have no idea who uses this sampleset and for what purpose. In my case, it is just for evening practice time for the Sunday services, and I love that I have the feeling of playing a real instrument as much as possible even on the screen. (The sound quality is even superior to our poor condition pipe organ… ☹) Others may use it for more serious purposes, they probably do not favour such a ‘fancy’, distracting interface. It is impossible to make everyone happy.

The left/right jamb is useful if one uses two touchscreens near the organ. This is a rare (although not absolutely uncommon) setup for GO. I have only one screen without touch, as I simply connected a low-cost USB synth and a DIY midified pedalboard (got the pedals free from a non-working pipe organ) to my desktop PC. (The whole system costed no more than 150€.) Others, who take it more seriously might benefit from such a new panel format.

**Comment by larspalo** [comment]
> It is impossible to make everyone happy.

Right you are!

I'm glad you like the sample set! I had the privilege to hold the organist position in Norrfjärden between 1998 and until I moved to Bureå in 2003. It's truly a marvelous instrument, even if it's somewhat difficult to play if you're only accustomed to modern type of instruments. But it sounds amazing! And once one gets accustomed to the touch, it's also capable of extremely expressive playing/articulations.

As long as you enjoy playing on the equipment you have, you have no reason to worry! I still use M-Audio keystations myself...

I'm in the finishing stages of updating the Bureå Church sample set and once I have that ready, I'll focus on updating Norrfjärden.

**Comment by larspalo** [comment, marked as answer]
Today, 20230618, I've updated the Norrfjärden Church Sample Set. It uses the same file name and the same church name as the previous package as it's expected to replace the older now obsolete version.

There are many improvements in this new version:

- Fixed the embedded pitch info in a few samples and corrected the tuning of the original temperament.
- It's now possible to actually play either the e-flat or d-sharp from the "Modern keyboards" that will visibly play the correct key on the original main panel keyboards. On the modern keyboards panel there is a switch for selecting either of the sub semi tones.
- D-sharp samples, or re-tuned versions, are added for the whole compass (not only in the original compass).
- The "Sperrventil" for each division now cancels all related sounds, even the D-sharp ones.
- Adjusted the release cue position for a few samples
- Added large stop jamb panels for both the left jamb and the right jamb.
- Adjusted the method of adding drawstop noises for most of the drawstops (that are not effect stops in themselves) so that the attack will sound independently from the release.
- On the "Modern keyboards" panel the labels for the drawstops are now actual labels with modern text instead of the original images seen on the main panel.
- The relatively new possibility to adjust PitchCorrection at the [Organ] level is used so that when re-tuning the instrument to a specific temperament the overall pitch will be more or less kept at original pitch even though the temperament changes. If this is not desired, adjust the pitch on the global settings panel to suite your taste.

As before, the Sample Set can be downloaded at http://familjenpalo.se/vpo/download/.

**Reply by kerkovits** [reply]
Thanks, it rocks!

**Comment by rousseldenis** [comment]
Great! I'll test it tonight.

@larspalo by the way, do you have advice to add some IR to the sample set? Thanks
